### ISSUE TITLE
Breaking Changes

### ISSUE THREAD

**Issue by Jonny-vb** [Open-CMSIS-Pack/STM32F4xx_DFP#1]
I just wanted to provide some feedback to the major changes introduced in the latest version of this pack.

This version of the pack breaks a lot of internal code that we've developed over the years in our own pack. We have conditions based on specific HAL components, which allows them to be brought in as needed which is great, and I thought a major selling point for the cmsis-packs in general.
Now we have to have a dependency on another companies tool, which forces lots of things about the project. If all we want to do is pull in the HAL or LL code, this is massive overkill. I can understand having it as an option for pin configurations (as has always been there), but forcing it to be used to pull in library code is just bad. Not to mention that the HAL files it gives you are not write protected like they would be if included directly from the cmsis-pack, making it quite easy for a developer to end up changing them accidently or on purpose.
This also means we can no longer look at the pack version to understand what version of the HAL is being used - we have to look into the files or into ST Cube to figure this out.

In short, we are not happy with the changes here. It felt like everything was going in the right direction with MDKv6 and cmsis-packs, but this is a major step backwards.
Using ST cube should be optional, not forced. And definitely not required to just get the HAL.

**Comment by jkrech** [comment]
Thanks a lot for sharing your feedback. I am sorry to hear that you are not happy with the changes and I do appreciate the reasons that you are listing.
I strongly suggest that you share your requirements and expectations with STMicroelectronics. I was told that they will distribute their own packs going forward and you should try to influence that.
In the current situation it was not sustainable to maintain DFPs for all ST Series redistributing a copy of the STCube FW Library and craft and maintain component descriptions. 
We strongly believe that it is a good approach to make use of STM32CubeMX and let it copy and process files which are then automatically copied into the projects.
